### ISSUE TITLE
Reading a Tensorflow 2.0's tfrecords File?

### ISSUE THREAD

**Issue by yourtheron** [tensorflow/tensorflow#35372]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

I have a .tfrecords file that stores my serialised data. How can I re-load it next time? It is impractical to generate it fro raw data each time when it is needed. Note that I tried TFRecordReader from [here](https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/t-f-record-reader) but it does not work at all

tf prompts

`
		AttributeError: module 'tensorflow' has no attribute 'TFRecordReader'
`

**Comment by ravikyram** [comment]
@yourtheron 
Will it be possible to share simple standalone code to reproduce the issue in our environment. It helps us in localizing the issue faster.

**Comment by yourtheron** [comment]
@ravikyram I found the root cause and the solution. In fact, if I want to write my serialised data into a TFRecord file, I cannot use `tensorflow::ops::TFRecordReader`. Instead, I have to use something like
```
with tf.io.TFRecordWriter(RecorderFileName) as writer:
```
and if I want to read data from the file, I have to firstly 'record' the file into a raw dataset by
```
trRawDataset = tf.data.TFRecordDataset(TrRecorderFileName)
```
Then read data into a training-ready dataset by `map()` function
```
dsTrain = trRawDataset.map(decodeMsg2Img)
```
where `decodeMsg2Img` is a function that uses `tf.io.parse_single_example` to parse the data as per features I gave when I wrote the data into the TFRecord file.

Thanks all the same! I am new to tf. It is quite a learning.
